src/server/main_service.cc

```diff
@@ -1047,6 +1047,39 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   }
 }
 
+#ifndef NDEBUG
+
+#define ENABLE_DCHECK_REPLIED                                                               \
```

Comment:
is it the only way to do it? by injecting a reply builder and bunch of preprocesor code?

Reply:
I injecting the CapturingReplyBuilder is clean enough and I cleaned up the preprocessor code. Do you have any better suggestions?